src/jit/codegenxarch.cpp

```diff
@@ -4477,78 +4599,107 @@ void CodeGen::genConsumeRegAndCopy(GenTree *tree, regNumber needReg)
-    // register type.
-    // This can happen if (currently iff) we have a SIMD vector type that fits in an integer
-    // register, in which case it is passed as an argument, or returned from a call,
-    // in an integer register and must be copied if it's in an xmm register.
+        GenTreeCopyOrReload* copyTree = treeNode->AsCopyOrReload();
+        GenTreeCall* call = op1->AsCall();
+        ReturnTypeDesc* retTypeDesc = call->GetReturnTypeDesc();
+        unsigned regCount = retTypeDesc->GetReturnRegCount();
 
-    bool srcFltReg = (varTypeIsFloating(op1)      || varTypeIsSIMD(op1));
-    bool tgtFltReg = (varTypeIsFloating(treeNode) || varTypeIsSIMD(treeNode));
-    if (srcFltReg != tgtFltReg)
-    {
-        instruction ins;
-        regNumber fpReg;
-        regNumber intReg;
-        if (tgtFltReg)
-        {
-            ins = ins_CopyIntToFloat(op1->TypeGet(), treeNode->TypeGet());
-            fpReg = targetReg;
-            intReg = op1->gtRegNum;
-        }
-        else
+        for (unsigned i = 0; i < regCount; ++i)
         {
-            ins = ins_CopyFloatToInt(op1->TypeGet(), treeNode->TypeGet());
-            intReg = targetReg;
-            fpReg = op1->gtRegNum;
+            var_types type = retTypeDesc->GetReturnRegType(i);
+            regNumber fromReg = call->GetRegNumByIdx(i);
+            regNumber toReg = copyTree->GetRegNumByIdx(i);
+            
+            // A Multi-reg GT_COPY node will have valid reg only for those
+            // positions that corresponding result reg of call node needs
+            // to be copied.
+            if (toReg != REG_NA)
+            {
+                assert(toReg != fromReg);
+                inst_RV_RV(ins_Copy(type), toReg, fromReg, type);
+            }
         }
-        inst_RV_RV(ins, fpReg, intReg, targetType);
     }
     else
     {
-        inst_RV_RV(ins_Copy(targetType), targetReg, genConsumeReg(op1), targetType);
-    }
+        var_types targetType = treeNode->TypeGet();
+        regNumber targetReg = treeNode->gtRegNum;
+        assert(targetReg != REG_NA);
 
-    if (op1->IsLocal())
-    {
-        // The lclVar will never be a def.
-        // If it is a last use, the lclVar will be killed by genConsumeReg(), as usual, and genProduceReg will
-        // appropriately set the gcInfo for the copied value.
-        // If not, there are two cases we need to handle:
-        // - If this is a TEMPORARY copy (indicated by the GTF_VAR_DEATH flag) the variable
-        //   will remain live in its original register.
-        //   genProduceReg() will appropriately set the gcInfo for the copied value,
-        //   and genConsumeReg will reset it.
-        // - Otherwise, we need to update register info for the lclVar.
+        // Check whether this node and the node from which we're copying the value have the same
+        // register type.
+        // This can happen if (currently iff) we have a SIMD vector type that fits in an integer
```

Comment:
I believe that this comment is worded incorrectly. I think you mean "Check whether this node ... have different register types. This can happen ..."